### PR TITLE
Restructure Migrations and Add Unit Tests

### LINKED ISSUE #27: Consolidate the Migrations into One
In visual studio delete all Migrations except for the Snapshot one and replace them with a single migration named initial schema to represent the default schema we used to implement all the background processes like admin and owner accounts as well as storage for DoodleNote images.

### PR DESCRIPTION
**Pull request by JB-coding02**
Closes #27 

Removed old migrations and replaced with a new initial migration for Identity and DoodleNotes schema. Updated ApplicationDbContextModelSnapshot. Added unit tests for controllers, services, and view models.

This pull request adds comprehensive unit tests for the admin accounts controller, role service, view models, and the doodle controller, significantly improving test coverage for the application's core features. Additionally, it removes obsolete or incompatible test and migration files, cleaning up the codebase.

**New Unit Tests:**

* **Admin features:**
  - Adds `AdminAccountsControllerTests` to verify controller responses for missing or invalid user and role data, ensuring correct HTTP status codes and error messages.
  - Adds `RoleServiceTests` to cover role assignment, promotion/demotion logic, and edge cases such as owner demotion and admin-only actions.

* **View model and controller tests:**
  - Adds `ViewModelTests` to validate computed properties, default values, and collection initialization in various view models.
  - Adds `DoodleControllerTests` to check that the `Index` action returns a `ViewResult`.

**Codebase Cleanup:**

* Removes an obsolete `ApplicationDbContextTests` file and the auto-generated EF Core migration designer file, eliminating outdated or incompatible code. [[1]](diffhunk://#diff-f1b097e9df94a4325c46892fc7815e0dc2bba129906402eba91e2157819685c1L1-L7) [[2]](diffhunk://#diff-62583f99281df4dc73137a7c31ddfeaefac186000f922acb5f251a7188708781L1-L277)

### CODE REVIEW

File: DoodleNote.Tests/AdminAccountsControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		IActionResult result = await controller.AssignRole("user-id", " ");
+
+		BadRequestObjectResult badRequest = Assert.IsType<BadRequestObjectResult>(result);
+		Assert.Equal("User ID and role are required.", badRequest.Value);
+	}
+
+	[Fact]
+	public async Task RemoveRole_ReturnsBadRequest_WhenRoleMissing()
+	{
+		AdminAccountsController controller = new(CreateUserManagerMock().Object, Mock.Of<RoleService>());
```
[the remark this answers is not in the document]
Comment (JB-coding02) [reply DoodleNote.Tests/AdminAccountsControllerTests.cs:55]: @copilot apply changes based on [this feedback](https://github.com/RayHamps03/DoodleNote/pull/28#discussion_r3243773892)

Refactor RoleService to implement an interface and Mock that interface
